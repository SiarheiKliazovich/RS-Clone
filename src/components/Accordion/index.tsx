import { FunctionComponent, useState } from "react";
import classNames from "classnames";
import { AccordionType } from "../../types/configurator";

const Accordion: FunctionComponent<AccordionType> = ({
  title,
  children,
}: AccordionType) => {
  const [active, setActive] = useState<boolean>(false);
  return (
    <>
      <div className="wizard-item__title" onClick={() => setActive(!active)}>
        {title}
      </div>
      <div
        className={classNames("form-color__wrapper", {
          active: active,
        })}
      >
        {children}
      </div>
    </>
  );
};
export default Accordion;
