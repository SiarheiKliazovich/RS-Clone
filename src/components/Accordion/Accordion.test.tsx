import { render } from "@testing-library/react";
import Accordion from ".";

describe("Accordion", () => {
  test("renders correctly", () => {
    const { container } = render(
      <Accordion title="Test Accordion">
        <p>Test Content</p>
      </Accordion>
    );
    expect(container.firstChild).toMatchSnapshot();
  });
});
