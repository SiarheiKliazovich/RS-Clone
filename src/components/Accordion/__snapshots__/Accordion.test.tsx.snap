// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Accordion renders correctly 1`] = `
<div
  class="wizard-item__title"
>
  Test Accordion
</div>
`;
